xdsl/interactive/app.py

```diff
@@ -389,24 +403,60 @@ def get_query_string(self) -> str:
         )
         return f"xdsl-opt -p {query}"
 
+    def get_operation_count_diff(self) -> tuple[tuple[str, int, int], ...]:
+        """
+        Function returning a tuple of tuples containing the diff of the input and output
+        operation name and count.
+        """
+        input_op_count_dict = dict(self.input_operation_count_tuple)
+        output_op_count_dict = dict(self.output_operation_count_tuple)
+        res = ()
+        # iterate through output_operation_count_tuple
+        for k, v in self.output_operation_count_tuple:
+            if k in input_op_count_dict:
+                diff = v - input_op_count_dict[k]
+                res = (
+                    *res,
+                    (k, v, diff),
+                )
+            else:
+                diff = v
+                res = (
+                    *res,
+                    (k, v, diff),
+                )
+
+        # iterate through input_operation_count_tuple
+        for k, v in input_op_count_dict.items():
+            if k not in output_op_count_dict:
+                diff = -input_op_count_dict[k]
+                res = (
+                    *res,
+                    (k, 0, diff),
+                )
+
+        return res
```

Comment:
I'd recommend something like this, to create a set of all keys, and iterate over that once, and accumulate into a dict also.

```suggestion
        input_op_count_dict = dict(self.input_operation_count_tuple)
        output_op_count_dict = dict(self.output_operation_count_tuple)
        all_keys = {*input_op_count_dict, *output_op_count_dict}
        
        res: dict[str, tuple[int, int]] = {}
        for k in all_keys:
            input_count = input_op_count_dict.get(k, 0)
            output_count = output_op_count_dict.get(k, 0)
            res[k] = (output_count, output_count - input_count)

        return tuple((k, v0, v1) for (k, (v0, v1)) in res.items().sorted())
```

Reply:
I would also recommend putting this logic into a function near the one that computes the dicts and to test the logic separately from the gui tool